client/rest/src/plugins/rosetta/nem/mempoolRoutes.js

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright (c) 2016-2019, Jaguar0625, gimre, BloodyRookie, Tech Bureau, Corp.
+ * Copyright (c) 2020-present, Jaguar0625, gimre, BloodyRookie.
+ * All rights reserved.
+ *
+ * This file is part of Catapult.
+ *
+ * Catapult is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Catapult is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Catapult.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import { OperationParser } from './OperationParser.js';
+import { createLookupCurrencyFunction, getBlockchainDescriptor } from './rosettaUtils.js';
+import MempoolResponse from '../openApi/model/MempoolResponse.js';
+import MempoolTransactionRequest from '../openApi/model/MempoolTransactionRequest.js';
+import MempoolTransactionResponse from '../openApi/model/MempoolTransactionResponse.js';
+import NetworkRequest from '../openApi/model/NetworkRequest.js';
+import TransactionIdentifier from '../openApi/model/TransactionIdentifier.js';
+import { RosettaErrorFactory, rosettaPostRouteWithNetwork } from '../rosettaUtils.js';
+import { NetworkLocator } from 'symbol-sdk';
+import { Network } from 'symbol-sdk/nem';
+
+export default {
+	register: (server, db, services) => {
+		const blockchainDescriptor = getBlockchainDescriptor(services.config);
+		const network = NetworkLocator.findByName(Network.NETWORKS, blockchainDescriptor.network);
+		const lookupCurrency = createLookupCurrencyFunction(services.proxy);
+		const parser = new OperationParser(network, {
+			includeFeeOperation: true,
+			operationStatus: 'success',
+			lookupCurrency
+		});
+
+		// since this is P2P API, simulate an authenticated request
+		const getUnconfirmedTransactions = () => services.proxy.fetch('transactions/unconfirmed', jsonObject => jsonObject.entity.data, {
+			method: 'POST',
+			headers: { 'Content-Type': 'application/json' },
+			body: JSON.stringify({
+				challenge: { data: 'AAAAAAAAAA' },
+				entity: { hashShortIds: [] }
+			})
+		});
+
+		server.post('/mempool', rosettaPostRouteWithNetwork(blockchainDescriptor, NetworkRequest, async () => {
+			const transactions = await getUnconfirmedTransactions();
+
+			const response = new MempoolResponse();
+			response.transaction_identifiers = transactions.map(transaction =>
+				new TransactionIdentifier(transaction.meta.hash.data.toUpperCase()));
```

Comment:
Don't see the meta property 🤔 
```
curl -H "Content-Type: application/json" -X POST "http://jusan.nem.ninja:7890/transactions/unconfirmed" -d '{ "entity": { "hashShortIds": [] }, "challenge": { data: "AAAAAAAAAA" } }' -s | jq
{
  "signature": "ec76a4a811856303b7243b4c020c88c4b28b09f13395432de7b2763c85421c1d08524612497e2803fa6f1563331f815e20c501e68dba1c38ab1cc0702bf43f01",
  "entity": {
    "data": [
      {
        "timeStamp": 295587662,
        "amount": 54392941,
        "signature": "46ee94eb9b4e1ab973733161582e3d91910acea21cf8226682ce080e7b6c8fd18dd4fc5d595a2e52086b5e67e420d77ffc4983e5ee13362828224481eb73300a",
        "fee": 50000,
        "recipient": "NCZFSBQ4ODZZP3ZCPVX2AD3SDDZBQRCQUTU4ZLTS",
        "type": 257,
        "deadline": 295602052,
        "message": {},
        "version": 1744830465,
        "signer": "237f5df4e1c827d4fca63731d7e97e630c26a8c45918bc3663d58bd6c3468e53"
      }
    ]
  }
}
```

Reply:
you're right ... it doesn't have hash, maybe we should update NIS?